packages/react-admin/src/EditDialog.tsx

```diff
@@ -45,10 +45,13 @@ export function useEditDialog(): [React.ComponentType<IProps>, { id?: string; mo
         [selectionApi],
     );
 
-    const api: IEditDialogApi = {
-        openAddDialog,
-        openEditDialog,
-    };
+    const api: IEditDialogApi = React.useMemo(() => {
```

Comment:
this memo has no benefit and can be removed

Reply:
48:11  warning  The 'api' object makes the dependencies of useMemo Hook (at line 61) change on every render. To fix this, wrap the initialization of 'api' in its own useMemo() Hook  react-hooks/exhaustive-dep

Either we remove the useMemo also for EditDialogWithHookProps or we wrap both in useMemo since EditDialogWithHookProps uses api as an dependency.

Reply:
ok, true, since openAddDialog and openEditDialog could be memoized too useMemo for api makes sense like it is!